app/src/main/java/ch/epfl/sdp/blindly/user/UserHelper.kt

```diff
@@ -134,55 +92,56 @@ class UserHelper {
-                       sexualOrientations: List<String>, showMe: String,
-                       passions: List<String>) {
+    fun getUserId(): String? {
+        return FirebaseAuth.getInstance().currentUser?.uid
+    }
 
+    /**
+     * Set User's information in firestore after user entered his information in set_profile
+     */
+    fun setUserProfile(userBuilder:User.Builder) {
         val database = Firebase.firestore
-
-        if (getUserId() != null) {
-            val information = arrayListOf(name,
-                location,
-                birthday,
-                genre,
-                sexualOrientations,
-                showMe,
-                passions,
-                DEFAULT_RADIUS)
-            val newUser = userFields.zip(information).toMap()
-
-            database.collection(USER_PATH).add(newUser)
-                    .addOnSuccessListener { documentReference ->
-                        Log.d(TAG, "User's profile was set: ${documentReference.id}")
+        val uid = getUserId()
+        if (uid != null) {
+            val newUser = userBuilder.setRadius(DEFAULT_RADIUS)
+                .setMatches(listOf())
+                .setDescription("")
+                .build()
+
+            database.collection(USER_COLLECTION).document(uid).set(newUser)
+                    .addOnSuccessListener {
+                        Log.d(TAG, "User \"$uid\" was set in firestore")
                     }
                     .addOnFailureListener { e ->
                         Log.w(TAG, "Error setting the user's profile", e)
                     }
         }
     }
 
-    /**
-     * field: the field of the value to change inside the database
-     * newValue: the new value to set for the user
-     */
-    fun <T> updateProfile(field: String, newValue: T) {
-        if(!userFields.contains(field))
-            throw IllegalArgumentException("Path must be one of $userFields")
-        if(newValue !is String || newValue !is ArrayList<*>)
-            throw IllegalArgumentException("Expected String or ArrayList<String>")
-        val database = Firebase.firestore
-        val user = getUserId()
-        if (user != null)
-            database.collection(USER_PATH)
-                    .document(user)
-                    .update(field, newValue)
+    fun setEmail(email: String): Task<Void>? {
+        return FirebaseAuth.getInstance().currentUser?.updateEmail(email)
+    }
+
+    fun getEmail(): String? {
+        return FirebaseAuth.getInstance().currentUser?.email
     }
 
+    fun updatePassword(password: String) {
+        val user = FirebaseAuth.getInstance().currentUser
+
+        user!!.updatePassword(password)
+                .addOnCompleteListener { task ->
+                    if (task.isSuccessful) {
+                        Log.d(Companion.TAG, "User password updated.")
+                    } else {
+                        Log.d(Companion.TAG, "Error: Could not update password.")
```

Comment:
Might be nice to have those messages in private constants.

Reply:
I'm not sure whether or not, exception and logs necessitate const

Reply:
Maybe not. Aftert thinking about it, I think we would have a lot of string constants for messages. Better keep it like this.